jumpscale/sals/nginx/nginx.py

```diff
@@ -114,26 +115,48 @@ def get_config(self):
 
     def generate_certificates(self):
         if self.ssl:
-            j.sals.process.execute(
+            res = j.sals.process.execute(
                 f"certbot --nginx -d {self.domain} --non-interactive --agree-tos -m {self.letsencryptemail} --nginx-server-root {self.parent.cfg_dir}"
             )
+            return res
+
+    def generate_self_signed_certificates(self):
+        if self.ssl:
+            self.selfsigned = True
+            if j.sals.fs.exists(f"{self.parent.cfg_dir}/key.pem") and j.sals.fs.exists(
+                f"{self.parent.cfg_dir}/cert.pem"
+            ):
+                return
+            res = j.sals.process.execute(
+                f"openssl req -nodes -x509 -newkey rsa:4096 -keyout {self.parent.cfg_dir}/key.pem -out {self.parent.cfg_dir}/cert.pem -days 365 -subj '/CN=localhost'"
+            )
+            return res
 
     def configure(self, generate_certificates=True):
         j.sals.fs.mkdir(self.cfg_dir)
-
         for location in self.get_locations():
             location.configure()
 
+        failback = self.selfsigned
+        self.selfsigned = False
         j.sals.fs.write_file(self.cfg_file, self.get_config())
-
-        if generate_certificates:
-            self.generate_certificates()
+        self.selfsigned = failback
+
+        if generate_certificates and self.ssl:
+            res = self.generate_certificates()
+            if res and res[0] != 0:
+                if failback:
+                    # failed to generate cert by certbot
+                    # use self_signed certs
+                    self.generate_self_signed_certificates()
```

Comment:
Here we assume this call always works without problems and ignoring the rc, out, err data